### ISSUE TITLE
Inconsistent list processing

### ISSUE THREAD

**Issue by szymonkups** [ckeditor/ckeditor5-markdown-gfm#11]
I did some further testing with lists conversion on GitHub and it looks like they're inserting paragraphs inside list items in some situations (so are libraries we use). We must be sure that view which is displayed in RTE will produce correct markdown output that later on will be converted by GitHub to HTML. And that HTML needs to look similar to the view presented in RTE. This is tricky sometimes.

### Example 1
Let's take this view:
```html
<ul><li>item</li></ul><ol><li>item</li></ol>
```
our data processor will convert this to:
```markdown
*   item

1.  item
```
But when we will convert it back to markdown, the result is:
```html
<ul><li><p>item</p></li><li><p>item</p></li></ul>
```
This should be definitely two lists instead of one. Question is: should we insert paragraphs inside those lists? Markdown from this example is converted by GitHub to:
```html
<ul>
<li><p>item</p></li>
</ul>

<ol>
<li>item</li>
</ol>
```
Should our view look the same, or we could skip paragraphs from the first list item and be a little inconsistent with what GitHub is doing? I believe that current list feature implementation doesn't allow to put paragraph inside list item.

### Example 2
When there is an empty line between list items, they are wrapped in paragraphs (same behavior on GitHub and our data processor)
```markdown
* one

* two
```
is processed to:
```html
<ul><li><p>one</p></li><li><p>two</p></li></ul>
```
The same issue as in **Example 1** - should we convert such lists to have paragraphs in `<li>` or not?

**Comment by scofalik** [comment]
I am not sure, but isn't it that we don't what DOM do we create, as long as it looks good/GitHub like?

Right now, in our model, it's invalid to have `paragraph`s in `listItem`s. I'd rather avoid the situation where we have `paragraph`s in `listItem`s because right now we treat them as same level elements. I don't know whether mixing them would not break some algorithms. I am for stripping `<p>`s if this has no impact on how the content looks in editable.

I am not sure whether I understand correctly what happens here, but if:

* content in editable looks fine, and
* markdown is correct

Then the solution is correct.

**Comment by Reinmar** [comment]
There's a golden rule that we always apply to such questions – if you can't create some content using the editor, don't worry how it's going to be preserved by the editor when you load it. This means that you should only worry about gracefully converting such content into correct content (matching the schema), which is important mainly for pasting. And that you must have parity between `getData()` and `setData()` because everything what was created in the editor must be also accepted by it.

In other words, the only thing that could worry us here is when `getData()` would output lists in such a way that `setData()` (or, in fact, the MdDP) would be confused and think that there are some paragraphs inside the list items.
